### ISSUE TITLE
As Pipes user I'd like to get more feedback from Pipes whether I have configured the blocks in my graph properly

### ISSUE THREAD

**Issue by josvanroosmalen** [marklogic-community/pipes#126]
In order to make Pipes more "dummy"/"fool" proof, I want to work on a sanity checker, and one of the first things is a sanity checker during editing the graphs, e.g. check that Feature/Lookup has a valid `cts.query`. 

During editing we need need to be able to validate properties/widgets/etc.  Preferable the callback should be able to return "false" and then the editor should not be closed, and we need to be able to provide feedback via alert. 

This ticket is not about implementing sanity checks itself, but only the callbacks. so that we can build sanity checking on this.

**Comment by sashamitrovich** [comment]
@josvanroosmalen, can we rephrase this as: "As Pipes user I'd like to get more feedback from Pipes whether I have configured the blocks in my graph properly"? Is this correct?

**Comment by josvanroosmalen** [comment]
yes

**Comment by danielholgate** [comment]
There are two different editing mechanisms in Pipes.  
One is the native litegraph property editing, which opens a small black box:
<img width="425" alt="Screen Shot 2020-03-31 at 14 43 45" src="https://user-images.githubusercontent.com/13762269/78027705-16829c80-735e-11ea-961d-b823390acc6f.png">

The second are the editing screens which open up for blocks (current only three) via double click and are actually a window in the Pipes application rather than litegraph:
<img width="863" alt="Screen Shot 2020-03-31 at 14 45 24" src="https://user-images.githubusercontent.com/13762269/78027812-46ca3b00-735e-11ea-8fc9-b677807d1bdc.png">

Capturing the second type of dialog is easy because are already Pipes code and can be captured during/before/after edit. The first type of property editing requires adding hooks into the litegraph framework

Need to confirm which properties are targeted in this ticket

**Comment by sashamitrovich** [comment]
> yes

Thanks, I changed the title.
